### PR TITLE
Test misk+protoc clients+servers in every direction

### PR DESCRIPTION
**Pull request by swankjesse**
This introduces a (bad) GrpcClient class to be replaced with something
that does the same work but behind nicer syntax.

### CODE REVIEW

File: misk-grpc-tests/src/main/kotlin/misk/grpc/protocserver/RouteGuideProtocServiceModule.kt [hunk 1]
```diff
@@ -1,13 +1,19 @@
 package misk.grpc.protocserver
 
 import com.google.common.util.concurrent.Service
+import com.google.inject.Provides
 import io.grpc.BindableService
 import misk.inject.KAbstractModule
+import javax.inject.Named
 
 /** A module that runs a standard gRPC server: generated protoc protos and a Netty backend. */
```
Comment (adrw) [inline misk-grpc-tests/src/main/kotlin/misk/grpc/protocserver/RouteGuideProtocServiceModule.kt]: Spelling nit: *Jetty ?

File: misk-grpc-tests/src/test/kotlin/misk/grpc/MiskClientMiskServerTest.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+class MiskClientMiskServerTest {
+  @MiskTestModule
+  val module = Modules.combine(
+      MiskGrpcClientModule(),
+      RouteGuideMiskServiceModule())
+
+  @Inject lateinit var grpcClientProvider: Provider<GrpcClient>
+
+  @Test
+  @DisabledOnJre(JAVA_8) // gRPC needs HTTP/2 which needs ALPN which needs Java 9+.
```
Comment (adrw) [inline misk-grpc-tests/src/test/kotlin/misk/grpc/MiskClientMiskServerTest.kt]: Nice 👍

File: misk-grpc-tests/src/main/kotlin/misk/grpc/protocserver/RouteGuideProtocServiceModule.kt (hunk 1 above)
Comment (swankjesse) [reply misk-grpc-tests/src/main/kotlin/misk/grpc/protocserver/RouteGuideProtocServiceModule.kt]: Nope!
https://netty.io/
https://www.eclipse.org/jetty/

Comment (adrw) [reply misk-grpc-tests/src/main/kotlin/misk/grpc/protocserver/RouteGuideProtocServiceModule.kt]: Oops!

**Pull request merged**